src/get-workspaces-paths.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const execa = require('execa');
+const { promisify } = require('util');
+const glob = promisify(require('glob'));
+const path = require('path');
+const readJson = require('./json').read;
+
+async function getWorkspacesPaths({
+  workspaceCwd,
+}) {
+  let rootWorkspacePackageJson = await readJson(path.join(workspaceCwd, 'package.json'));
+
+  let { workspaces } = rootWorkspacePackageJson;
+
+  let packagesGlobs;
+
+  let _1dFilesArray;
+
+  if (!workspaces) {
+    _1dFilesArray = (await execa('pnpm', ['recursive', 'exec', '--', 'node', '-e', 'console.log(process.cwd())'], { cwd: workspaceCwd })).stdout
+      .split(/\r?\n/)
+      .map(workspace => path.relative(workspaceCwd, workspace));
+
+    packagesGlobs = [];
+  } else {
+    packagesGlobs = workspaces.packages || workspaces;
+    
+    let _2dFilesArray = await Promise.all(packagesGlobs.map(packagesGlob => {
+      return glob(packagesGlob, {
+        cwd: workspaceCwd,
+      });
+    }));
+    
+    _1dFilesArray = Array.prototype.concat.apply([], _2dFilesArray);
+  }
+    
+  workspaces = [...new Set(_1dFilesArray)];
+
+  let packageDirs = workspaces.map(dir => path.join(workspaceCwd, dir));
+    
+  return { packageDirs, packagesGlobs, rootWorkspacePackageJson, workspaces };
```

Comment:
When is `workspaces` used?

Reply:
I don't think we need this until it is exposed as public API.

Reply:
ok, updated